### ISSUE TITLE
Can't cancel KucoinFutures orders: "Order is in not cancelable state"

### ISSUE THREAD

**Issue by RoscoeTheDog** [ccxt/ccxt#14096]
I'm implementing a DCA strategy bot which runs fine so far on binance. I'm working on futures now via KucoinFutures exchange. I can open a position without problems with a base limit order. The market order types don't seem to return information like the average price for all contracts bought within the order (if someone knows a fix to this let me know). 

Here's the bug:
After the DCA buy order is filled, I'll place a limit sell (_i.e_ 'takeprofit') order and save the returning order response id info. After each subsequent order as it buys and averages the position down, it'll reposition the take profit by trying to cancel the placed order via targeting the order response id.

I have verified after stepping through the debugger that the order is not being overwritten before reaching my `exchange.cancel_order()` method. However when canceling the order, I get the following error:

`{Bad Request} kucoinfutures Order is in not cancelable state`

Clearly the request can see the order but it can't interact with it or close it. On the exchange, I can see it still exists and is active without anything filled. It does not specify _why_ it cannot interacted with it and the returning resulting information from the kucoinfutures api seems kind of minimal.

How can I cleanly cancel an opened futures limit order on the exchange? Are there any other limitations from the kucoinfutures endpoint that could prevent me from implimenting said bot?

**Comment by samgermain** [comment]
> I'm implementing a DCA strategy bot which runs fine so far on binance. I'm working on futures now via KucoinFutures exchange. I can open a position without problems with a base limit order. The market order types don't seem to return information like the average price for all contracts bought within the order (if someone knows a fix to this let me know).
> 
> Here's the bug: After the DCA buy order is filled, I'll place a limit sell (_i.e_ 'takeprofit') order and save the returning order response id info. After each subsequent order as it buys and averages the position down, it'll reposition the take profit by trying to cancel the placed order via targeting the order response id.
> 
> I have verified after stepping through the debugger that the order is not being overwritten before reaching my `exchange.cancel_order()` method. However when canceling the order, I get the following error:
> 
> `{Bad Request} kucoinfutures Order is in not cancelable state`
> 
> Clearly the request can see the order but it can't interact with it or close it. On the exchange, I can see it still exists and is active without anything filled. It does not specify _why_ it cannot interacted with it and the returning resulting information from the kucoinfutures api seems kind of minimal.
> 
> How can I cleanly cancel an opened futures limit order on the exchange? Are there any other limitations from the kucoinfutures endpoint that could prevent me from implimenting said bot?

Could you share a more complete version of your code please? Something I could run myself and test out?
I don't know exactly what you mean by ` the order is not being overwritten before reaching my exchange.cancel_order() method`

**Comment by RoscoeTheDog** [comment]
> > I'm implementing a DCA strategy bot which runs fine so far on binance. I'm working on futures now via KucoinFutures exchange. I can open a position without problems with a base limit order. The market order types don't seem to return information like the average price for all contracts bought within the order (if someone knows a fix to this let me know).
> > Here's the bug: After the DCA buy order is filled, I'll place a limit sell (_i.e_ 'takeprofit') order and save the returning order response id info. After each subsequent order as it buys and averages the position down, it'll reposition the take profit by trying to cancel the placed order via targeting the order response id.
> > I have verified after stepping through the debugger that the order is not being overwritten before reaching my `exchange.cancel_order()` method. However when canceling the order, I get the following error:
> > `{Bad Request} kucoinfutures Order is in not cancelable state`
> > Clearly the request can see the order but it can't interact with it or close it. On the exchange, I can see it still exists and is active without anything filled. It does not specify _why_ it cannot interacted with it and the returning resulting information from the kucoinfutures api seems kind of minimal.
> > How can I cleanly cancel an opened futures limit order on the exchange? Are there any other limitations from the kucoinfutures endpoint that could prevent me from implimenting said bot?
> 
> Could you share a more complete version of your code please? Something I could run myself and test out? I don't know exactly what you mean by ` the order is not being overwritten before reaching my exchange.cancel_order() method`

That would take me some time to translate everything unless you wanted to deal with a bunch of dependencies and setup a web server dev environment. I am using the Django framework and an sqlite3 database to handle the messaging from a user to control trade bots. For reliabilities sake and an extra layer of redundancy, I am saving all deal variables into a table in the DB so if anything fails or goes wrong I can restore the active trade deal's state from where it left off and do other debugging. There are a lot of checks and guards in my code for specific types and everything is constantly updating the record from the Table. For this reason my code isn't exactly portable.

If you can't proceed without further information then just give me a day or so to make a pseudo example. I'm also healing from an injury so I take breaks often. The TLDR is the output order id from the initial order was the same as the passed parameter id when trying to get the order from the exchange using `cancel_order()`. The resulting return from `cancel_order()` is `{Bad Request} kucoinfutures Order is in not cancelable state` every time. I jumped through the debugger to confirm the ID's upon each invocation. To place the order I was using the `place_order()` method, but I believe I also tried `place_market_order()` and `place_limit_order()` without further progress either.

**Comment by RoscoeTheDog** [comment]
Well when starting a new project and trying to do the same thing it worked. I checked the ccxt version in the original project and it was only a bit out dated. When I updated it it surprisingly solved the issue. 

There still remains a problem where the order response returns `None` for everything besides a few parameters like the `id` (at least on the Kucoin exchange). I'm going to close this issue as solved for now but if you have any answer to the `None` thing then feel free to open and comment on that as well.
